### DISCUSSION TITLE
Updating letsencrypt certs

### DISCUSSION THREAD

**Discussion by ejheil** [narrowlink/narrowlink#104] [Q&A] [answered]
I didn't see anything in the docs about this... I set up and forgot about letsencrypt a while ago; it's WebPublishing stuff on a domain and several subdomains.  It got letsencrypt certs for them automatically, but I'm now getting notifications from LetsEncrypt that they'll expire soon. Do I need to do something to renew the certs narrowlink generated?  If so, what?  Thanks!

**Comment by SajjadPourali** [comment, marked as answer]
Hello, it automatically renews the certificates when they expire in 7 days or less.

https://github.com/narrowlink/narrowlink/blob/48d2c8e549fff482b7b2a0533b9172fb61dff500/gateway/src/service/certificate/mod.rs#L105-L127

**Reply by ejheil** [reply]
Oh great, thank you!  I'm still 19 days out from expiration, so it makes sense they haven't done anything yet.

**Reply by SajjadPourali** [reply]
You're most welcome.
